Makefile

```diff
@@ -9,8 +24,47 @@ GO_FILES ?= $$(find . -name '*.go' | grep -v generated)
 	@./.dapper.tmp -v
 	@mv .dapper.tmp .dapper
 
+.nodapper:
+	$(info Checking essential build tools.)
+	@if [ ! -d $(WD)/bin ] ; then \
+		mkdir $(WD)/bin ; \
+	fi
+	$(info Checking go version for compatibility.)
+	@if [ ! -d $(GOROOT) ] ; then \
+		echo "No go found, fetching compatible version." ; curl -sL https://go.dev/dl/go$(GO_VERSION).linux-$(ARCH).tar.gz | tar -C $$PWD/bin -zxf - ; \
+	else \
+		case "$$(go version)" in \
+			*$(GO_VERSION)* ) echo "Compatible go version found." ;; \
+			* ) echo "Go appears to be " $$(go version) ; echo "Incompatible or non-functional go found, fetching compatible version." ; curl -sL https://go.dev/dl/go$(GO_VERSION).linux-$(ARCH).tar.gz | tar -C $$PWD/bin -zxf - ;; \
+		esac \
+	fi
+	@if ! type yq 2>/dev/null ; then \
+		echo "yq not found, fetching."; \
```

Comment:
could we add `yq` to Build-Depends in debian/control so we don't need to fetch it?

Reply:
Agreed with your comments until I did the tests and surprisingly there are 2 implementation of 'yq' and  debian is not using the one that k3s is using - https://github.com/mikefarah/yq), but instead is using the implementation from https://github.com/kislyuk/yq , hence the error that I'm seeing:
`usage: yq [-h] [--yaml-output] [--yaml-roundtrip]
          [--yaml-output-grammar-version {1.1,1.2}] [--width WIDTH]
          [--indentless-lists] [--in-place] [--version]
          [jq_filter] [files ...]
yq: error: argument files: can't open '.dependencies[] | select(.name == "golang: upstream version").version': [Errno 2] No such file or directory: '.dependencies[] | select(.name == "golang: upstream version").version'
`
My understanding is that the debian version is older and not so widely used. I don't think that we should adapt the script for the older version, but instead keep the modern 'yq' that supports the syntax already.